app/src/main/java/org/wikipedia/analytics/eventplatform/ImageRecommendationsEvent.kt

```diff
@@ -0,0 +1,60 @@
+package org.wikipedia.analytics.eventplatform
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.wikipedia.WikipediaApp
+import org.wikipedia.descriptions.DescriptionEditActivity
+import org.wikipedia.util.ActiveTimer
+import java.net.URLEncoder
+
+@Serializable
+@SerialName("/analytics/mobile_apps/android_image_recommendation_event/1.0.0")
+class ImageRecommendationsEvent(
+   private val action: String,
+   private val active_interface: String,
+   private val action_data: String,
+   private val primary_language: String,
+   private val wiki_id: String
+) : MobileAppsEvent(STREAM_NAME) {
+
+    companion object {
+        private const val STREAM_NAME = "android.android_image_recommendation_event"
+        val reasons = listOf("notrelevant", "noinfo", "offensive", "lowquality", "unfamiliar", "other")
+        private val timer = ActiveTimer()
+
+        fun logImpression(activeInterface: String, actionData: String = "", wikiId: String = "") {
+            if (activeInterface == "recommendedimagetoolbar") {
+                timer.reset()
+            }
+            submitImageRecommendationEvent("impression", activeInterface, actionData, wikiId)
+        }
+
+        fun logAction(action: String, activeInterface: String, actionData: String = "", wikiId: String = "") {
+            submitImageRecommendationEvent(action, activeInterface, actionData, wikiId)
+        }
+
+        fun logSeEditSuccess(action: DescriptionEditActivity.Action, wikiId: String, revisionId: Long) {
+            when (action) {
+                DescriptionEditActivity.Action.ADD_DESCRIPTION -> logAction("edit_success", "se_add_description", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION -> logAction("edit_success", "se_translate_description", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.ADD_CAPTION -> logAction("edit_success", "se_add_caption", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.TRANSLATE_CAPTION -> logAction("edit_success", "se_translate_caption", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                else -> logAction("edit_success", "se_add_image_tags", getActionDataString(revisionId = revisionId.toString()), wikiId)
+            }
+        }
+
+        fun getActionDataString(filename: String? = null, recommendationSource: String? = null,
+                                recommendationSourceProjects: String? = null, rejectionReasons: String? = null,
+                                acceptanceState: String? = null, revisionId: String? = null, captionAdd: Boolean? = null,
+                                altTextAdd: Boolean? = null, addTimeSpent: Boolean = false): String {
```

Comment:
Would it be possible to have the parameters have a better format since this function has a lot of parameters? For example:
```
                               (filename: String? = null,
                                recommendationSource: String? = null,
                                recommendationSourceProjects: String? = null,
                                rejectionReasons: String? = null,
                                acceptanceState: String? = null,
                                revisionId: String? = null,
                                captionAdd: Boolean? = null,
                                altTextAdd: Boolean? = null,
                                addTimeSpent: Boolean = false)
```